src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -160,7 +164,12 @@ private void setFilingApiData(
                             String.format("Empty submission returned when generating filing for %s", overseasEntityId)
                     ));
 
-    if (submissionDto.isForUpdate()) {
+    if (Boolean.TRUE.equals(submissionDto.isRemoveEntity())) {
+      var updateSubmission = new UpdateSubmission();
+      collectUpdateSubmissionData(updateSubmission, submissionDto, passThroughTokenHeader, true, logMap);
```

Comment:
I think 'is no change' flag will need to be set correctly here, like for update. User may say they are removing the OE and may either have updates to apply or no updates.

This will all come out of requirements though...

Reply:
Yes, not sure what "setting correctly" means for this: if the user is removing, then that is the change and so will always be true regardless, any other change alongside this does not make any sense.